trainee_template/ablation.amlg

```diff
@@ -0,0 +1,173 @@
+                    details (assoc
+                        influential_cases (true)
+                    )
+                    use_case_weights use_case_weights
+                    weight_feature weight_feature
+
+                ))
+                "influential_cases"
+            )
+        ))
+    )
+
+    ;used to compute and store the influence weight entropies for call cases contained in the Trainee in #Analyze
+    ; context_features: list of context features to use when determining influential cases.
+    #ComputeAndStoreInfluenceWeightEntropies
+    (let 
+        (assoc
+            influence_weight_entropy_map
+                ||(map
+                    (lambda
+                        (call ComputeInfluenceWeightEntropyUsingCaseID (assoc
+                            case_id (current_index 1)
+                            context_features context_features
+                        ))
+                    )
+                    (zip (call AllCases))
+                )
+            influence_weight_entropy_label ".influence_weight_entropy"
+        )
+
+        (call StoreCaseValues (assoc
+            case_values_map influence_weight_entropy_map
+            label_name influence_weight_entropy_label
+        ))
+
+        (assign_to_entities (assoc hasInfluenceWeightEntropies (true)))
+    )
+
+    ;initialize parameters and internal features related to auto ablation
+    ; weight_feature: optional, default '.case_weight'. name of feature whose values to use as case weights
+    #InitializeAutoAblate
+    (declare
+        (assoc
+            weight_feature ".case_weight"
+        )
+
+        (if (not hasPopulatedCaseWeight)
+            (seq
+                (assign_to_entities (assoc hasPopulatedCaseWeight (true)))
+                (call CreateCaseWeights (assoc
+                    feature_name weight_feature
+                ))
+            )
+        )
+    )
+
+    ;determine whether a new case (one that is not in the Trainee) should be ablated (trained as weights) or kept.
+    ; context_values: list of context values to react to
+    ; context_features: list of context features to react to
+    #ShouldNewCaseBeAblated
+    (declare
+        (assoc
+            context_features (list)
+            context_values (list)
+
+            ;internal
+            max_influence_weight_entropy_to_keep .infinity
+        )
+
+        ;If we do not have influence weight entropies stored, ablation cannot happen.
+        ; So, always return 1 when that is the case.
+        (if
+            (not (and hasInfluenceWeightEntropies))
+            (conclude 1)
+        )
+
+        (assign (assoc
+            max_influence_weight_entropy_to_keep
+                (compute_on_contained_entities (list
+                    (query_exists ".influence_weight_entropy")
+                    (query_quantile ".influence_weight_entropy" autoAblateEntropyThresholdQuantile ".case_weight")
+                ))
+        ))
+
+        (if
```

Comment:
if you wrap this 'or' in a (+) it'll numerify the boolean, so a (true) -> 1 and (false) -> 0, thus simplifying the code.  Just add a comment:
```
(+ (or 
 .... your code
 ))
```